spec/prog/vnet/rekey_nic_spec.rb

```diff
@@ -0,0 +1,206 @@
+# frozen_string_literal: true
+
+RSpec.describe Prog::Vnet::RekeyNic do
+  subject(:nx) {
+    described_class.new(st)
+  }
+
+  let(:st) { Strand.new }
+  let(:ps) {
+    PrivateSubnet.create_with_id(name: "ps", location: "hetzner-hel1", net6: "fd10:9b0b:6b4b:8fbb::/64",
+      net4: "1.1.1.0/26", state: "waiting")
+  }
+  let(:tunnel) {
+    n_src = Nic.create_with_id(private_subnet_id: ps.id,
+      private_ipv6: "fd10:9b0b:6b4b:8fbb:abc::",
+      private_ipv4: "10.0.0.1",
+      mac: "00:00:00:00:00:00",
+      encryption_key: "0x736f6d655f656e6372797074696f6e5f6b6579",
+      name: "default-nic")
+    n_dst = Nic.create_with_id(private_subnet_id: ps.id,
+      private_ipv6: "fd10:9b0b:6b4b:8fbb:def::",
+      private_ipv4: "10.0.0.2",
+      mac: "00:00:00:00:00:00",
+      encryption_key: "0x736f6d655f656e6372797074696f6e5f6b6579",
+      name: "default-nic")
+    IpsecTunnel.create_with_id(src_nic_id: n_src.id, dst_nic_id: n_dst.id).tap { _1.id = "0a9a166c-e7e7-4447-ab29-7ea442b5bb0e" }
+  }
+
+  before do
+    nx.instance_variable_set(:@nic, tunnel.src_nic)
+  end
+
+  describe ".sshable_cmd" do
+    let(:sshable) { instance_double(Sshable) }
+    let(:vm) {
+      vmh = instance_double(VmHost, sshable: sshable)
+      instance_double(Vm, vm_host: vmh)
+    }
+
+    it "returns vm_host sshable of source nic" do
+      expect(nx.nic).to receive(:vm).and_return(vm)
+      expect(sshable).to receive(:cmd).with("echo hello")
+      nx.sshable_cmd("echo hello")
+    end
+  end
```

Comment:
This is kind of an odd utility-level test that doesn't seem like it has much to do with nics per se, why's it in here?

Reply:
Oh I see, that's an interesting name for the symbol, and I see why you made it.